ballerina/channel_csv_io.bal

```diff
@@ -40,30 +40,89 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
+isolated function channelWriteCsv(string path, FileWriteOption option, string[][]|map<anydata>[] content) returns Error? {
+
+    WritableCSVChannel csvChannel;
     if content is string[][] {
+        csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = option));
         foreach string[] r in content {
             Error? writeResult = csvChannel.write(r);
             if writeResult is Error {
                 check csvChannel.close();
                 return writeResult;
             }
         }
+        check csvChannel.close();
     } else if content is map<anydata>[] {
+        string[] headers = [];
+        if content.length() == 0 {
+            GenericError e = error GenericError("Input contains an empty array.");
+            return e;
```

Comment:
Lets give specific error messages. Btw why is this error? For empty file, what is the expected behaviour?

Reply:
This error is given for the scenario where the user inputs a empty Record array. We can change this to `Input to the function contains an empty map<anydata>[]`